components/sidebar/sidebar_channel/sidebar_channel_menu/sidebar_channel_menu.tsx

```diff
@@ -1,284 +1,502 @@
+                                })
+                        }
+                        leadingElement={<i className='icon-close'/>}
+                    />,
+                ]),
+        ],
+    };
 
-        return (
-            <React.Fragment>
-                <Menu.Group>
-                    {markAsRead}
-                    {favorite}
-                    {muteChannel}
-                </Menu.Group>
-                <CategoryMenuItems
-                    channel={channel}
-                    openUp={this.state.openUp}
-                    location={'sidebar'}
-                />
-                <Menu.Group>
-                    {copyLink}
-                    {addMembers}
-                </Menu.Group>
-                {leaveChannel}
-            </React.Fragment>
+    //****************** Submenu items ******************//
+
+    let filteredCategories = categories?.filter(
+        (category) => category.type !== CategoryTypes.DIRECT_MESSAGES,
+    );
+
+    if (location === 'sidebar') {
+        const selectedChannels =
+            multiSelectedChannelIds.indexOf(channel.id) === -1 ?
+                [channel] :
+                displayedChannels.filter(
+                    (c) => multiSelectedChannelIds.indexOf(c.id) !== -1,
+                );
+        const allChannelsAreDMs = selectedChannels.every(
+            (selectedChannel) =>
+                selectedChannel.type === Constants.DM_CHANNEL ||
+                selectedChannel.type === Constants.GM_CHANNEL,
+        );
+        const allChannelsAreNotDMs = selectedChannels.every(
+            (selectedChannel) =>
+                selectedChannel.type !== Constants.DM_CHANNEL &&
+                selectedChannel.type !== Constants.GM_CHANNEL,
         );
-    }
 
-    handleOpenDirectionChange = (openUp: boolean) => {
-        this.setState({
-            openUp,
+        filteredCategories = categories?.filter((category) => {
+            if (allChannelsAreDMs) {
+                return category.type !== CategoryTypes.CHANNELS;
+            } else if (allChannelsAreNotDMs) {
+                return category.type !== CategoryTypes.DIRECT_MESSAGES;
+            }
+            return true;
         });
     }
 
-    onToggleMenu = (open: boolean) => {
-        this.props.onToggleMenu(open);
+    const categoryMenuItems = filteredCategories?.map(
+        (category: ChannelCategory) => {
+            return (
+                <MenuItem
+                    id={`moveToCategory-${channel.id}-${category.id}`}
+                    key={category.id}
+                    onClick={() => moveToCategory(category.id)}
+                    label={category.display_name}
+                    leadingElement={
+                        category.type === CategoryTypes.FAVORITES ? (
+                            <i className='icon-star-outline'/>
+                        ) : (
+                            <i className='icon-folder-outline'/>
+                        )
+                    }
```

Comment:
define this before the return in a `const` and pass it into the prop. That way we can reduce re-renders